### PR TITLE
[quest] Fix 'Arcane Legacy' Objectives

### LINKED ISSUE #6139: Quest Issue for quest 25766 - Arcane Legacy (Stonetalon Mountains, Cataclysm Classic)
## Description
Missing quest objective markers on the map/minimap for the required mobs. Quest Item is required to be used on these mobs to complete the quest.

## Wowhead link
Quest: https://www.wowhead.com/quest=25766/arcane-legacy

Mobs that should show up: https://www.wowhead.com/npc=34938/highborne-citizen
https://www.wowhead.com/npc=34932/highborne-sorcerer
https://www.wowhead.com/npc=34931/highborne-charlatan

## Questie & WoW version
Questie v10.3.5 & Cataclysm Classic

### PR DESCRIPTION
**Pull request by Sommos**
## Issue references

Fixes #6139

## Proposed changes

- Add a fix to include correct objectives for Arcane Legacy

### CODE REVIEW

File: Database/Corrections/cataQuestFixes.lua [hunk 1]
```diff
@@ -3247,6 +3247,9 @@ function CataQuestFixes.Load()
             [questKeys.objectives] = {{{41224,nil,Questie.ICON_TYPE_EVENT}}},
             [questKeys.extraObjectives] = {{nil,Questie.ICON_TYPE_EVENT,l10n("Disable the Shadow Cloak Generator"),0,{{"object",203208}}}},
         },
+        [25766] = { -- Arcane Legacy
+            [questKeys.objectives] = {{{34931,nil,Questie.ICON_TYPE_EVENT},{34932,nil,Questie.ICON_TYPE_EVENT},{34938,,nil,Questie.ICON_TYPE_EVENT}}},
```
Comment (BreakBB) [inline Database/Corrections/cataQuestFixes.lua:3251]:
```suggestion
            [questKeys.objectives] = {{{34931,nil,Questie.ICON_TYPE_EVENT},{34932,nil,Questie.ICON_TYPE_EVENT},{34938,nil,Questie.ICON_TYPE_EVENT}}},
```

Comment (Sommos) [reply Database/Corrections/cataQuestFixes.lua]: https://github.com/Questie/Questie/pull/6151/commits/d02f9f1edce67a3fef984a89fda6c8a26b03adce

**Comment by BreakBB** [comment]
On top of the additional `,` there is a thing about the actual fix:

As you can see on the [wowhead page for this quest](https://www.wowhead.com/quest=25766/arcane-legacy), the quest has a **single** objective, which is linked to an NPC. Therefore the objective in the game is of type "monster".

Now you added **multiple** monster objectives (as if the quest needs to you kill multiple different NPCs).

What we actually need for this quest is a `killCreditObjective`, which represents a monster objective for "kill any of these NPCs" (I know it is not "kill", but "use an item", but Questie doesn't care except for the icon to use).

This is the database representation of the `quest.objectives` field:

```lua
    ['objectives'] = 10, -- table
        --['creatureObjective'] = 1, -- table {{creature(int), text(string)},...}, If text is nil the default "<Name> slain x/y" is used
        --['objectObjective'] = 2, -- table {{object(int), text(string)},...}
        --['itemObjective'] = 3, -- table {{item(int), text(string)},...}
        --['reputationObjective'] = 4, -- table: {faction(int), value(int)}
        --['killCreditObjective'] = 5, -- table: {{creature(int), ...}, baseCreatureID, baseCreatureText}
        --['spellObjective'] = 6, -- table: {{spell(int), text(string), item(int)},...}
```

So instead of:

```lua
[questKeys.objectives] = {{{<npcId1>},{<npcId2>}}},
```

we need


```lua
[questKeys.objectives] = {nil,nil,nil,nil,{{{<npcId1>,<npcId2>,...},<npcId1>,"Some objective text"}}},
```

**Comment by BreakBB** [comment]
Oh and also `Questie.ICON_TYPE_INTERACT` is the correct one. Nice addition to directly add the override icon 👌🏻

**Review by BreakBB — APPROVED** [review]

**Pull request merged**